pyodide-build/pyodide_build/common.py

```diff
@@ -375,62 +375,37 @@ def init_environment(*, quiet: bool = False) -> None:
     Initialize Pyodide build environment.
     This function needs to be called before any other Pyodide build functions.
     """
-    if os.environ.get("__LOADED_PYODIDE_ENV"):
+
+    # Already initialized
+    if "PYODIDE_ROOT" in os.environ:
         return
 
-    os.environ["__LOADED_PYODIDE_ENV"] = "1"
+    try:
+        root = search_pyodide_root(Path.cwd())
+    except FileNotFoundError:  # Not in Pyodide tree
+        root = _init_xbuild_env(quiet=quiet)
 
-    _set_pyodide_root(quiet=quiet)
+    os.environ["PYODIDE_ROOT"] = str(root)
 
 
-def _set_pyodide_root(*, quiet: bool = False) -> None:
+def _init_xbuild_env(*, quiet: bool = False) -> Path:
     """
-    Set PYODIDE_ROOT environment variable.
-
-    This function works both in-tree and out-of-tree builds:
-    - In-tree builds: Searches for the root of the Pyodide repository in parent directories
-    - Out-of-tree builds: Downloads and installs the Pyodide build environment into the current directory
-
-    Note: this function is supposed to be called only in init_environment(), and should not be called directly.
+    Initialize the build environment for out-of-tree builds.
 
     Parameters
     ----------
     quiet
         If True, do not print any messages
     """
-
     from . import install_xbuildenv  # avoid circular import
 
-    # If we are building docs, we don't need to know the PYODIDE_ROOT
-    if "sphinx" in sys.modules:
-        os.environ["PYODIDE_ROOT"] = ""
-        return
```

Comment:
Don't we want to keep some version of this? I guess `search_pyodide_root` just works? why did we add this case in the first place?

Reply:
> why did we add this case in the first place?

If I remember correctly, we needed this because the `sphinx_argparse_cli` extension was required to run the pyodide-build source code to create the docs of pyodide-build CLI. But we removed pyodide-build CLI from docs in favor of pyodide CLI, it is not needed anymore.

Reply:
Ok, thanks for the explanation.